spec/support/helpers/model_spec_helpers.rb

```diff
@@ -0,0 +1,88 @@
+    let!(:q31) { create_question(parent: q3, name: "q31", type: "string", required: true, override: false) } # answered
+    let!(:q32) { create_question(parent: q3, name: "q32", type: "boolean", required: true) } # answered
+    let!(:q33) { create_group(parent: q3, name: "q33", required: true, position: 99) }
+    let!(:q331) { create_question(parent: q33, name: "q331", type: "boolean", required: false) } # answered
+    let!(:q332) { create_question(parent: q33, name: "q332", type: "boolean", loan_types: [lt2]) }
+    let!(:q34) { create_question(parent: q3, name: "q34", type: "string", required: false) }
+    let!(:q35) { create_question(parent: q3, name: "q35", type: "string", required: true, position: -99) }
+    let!(:q36) { create_question(parent: q3, name: "q36", type: "string", required: true, status: 'inactive') }
+    let!(:q37) { create_question(parent: q3, name: "q37", type: "string", required: true, status: 'retired') } # answered
+    let!(:q38) { create_group(parent: q3, name: "q38", required: false) }
+    let!(:q381) { create_question(parent: q38, name: "q381", type: "boolean", loan_types: []) }
+    let!(:q39) { create_question(parent: q3, name: "q39", type: "text", status: 'inactive') } # answered
+
+    # Optional group
+    let!(:q4) { create_group(parent: root, name: "q4", required: false) }
+    let!(:q41) { create_question(parent: q4, name: "q41", type: "string", required: false) }
+    let!(:q42) { create_question(parent: q4, name: "q42", type: "boolean", required: false) } # answered
+    let!(:q43) { create_question(parent: q4, name: "q43", type: "string", required: true) }
+    let!(:q44) { create_question(parent: q4, name: "q44", type: "string", required: true, override: false) }
+
+    # Inactive group
+    let!(:q5) { create_group(parent: root, name: "q5", required: true, status: 'inactive') }
+    let!(:q51) { create_question(parent: q5, name: "q51", type: "string", required: true) } # answered
+    let!(:q52) { create_question(parent: q5, name: "q52", type: "boolean", required: false) }
+
+    # Retired group
+    let!(:q6) { create_group(parent: root, name: "q6", required: true, status: 'retired') }
+    let!(:q61) { create_question(parent: q6, name: "q61", type: "string", required: false) } # answered
+    let!(:q62) { create_question(parent: q6, name: "q62", type: "boolean", required: true, status: 'inactive') }
+
+    before do
+      rset.set_response("q1", {"text" => "foo"})
+      rset.set_response("q2", {"text" => ""}) # required
+      rset.set_response("q31", {"text" => "junk"}) # required
+      rset.set_response("q32", {"boolean" => "no"}) # required
+      rset.set_response("q331", {"boolean" => "yes"})
+      rset.set_response("q37", {"text" => "retired question"})
+      rset.set_response("q39", {"text" => "inactive question"})
+      rset.set_response("q41", {"text" => ""})
+      rset.set_response("q42", {"text" => "pants"})
```

Comment:
I'm glad `pants` survived this refactor.

Reply:
😆